includes/RottenLinks.php

```diff
@@ -9,19 +9,34 @@ public static function getResponse( $url ) {
 		$config = $services->getConfigFactory()->makeConfig( 'rottenlinks' );
 
 		// Make the protocol lowercase
-		$urlexp = explode( '://', $url );
+		$urlexp = explode( '://', $url, 2 );
 		$proto = strtolower( $urlexp[0] ) . '://';
 		$site = $urlexp[1];
 		$urlToUse = $proto . $site;
 
+		$status = static::getHttpStatus( $urlToUse, 'HEAD', $services, $config );
+		// Some websites return 4xx or 5xx on HEAD requests but GET with the same URL gives a 200.
+		if ($status >= 400) {
+			$status = static::getHttpStatus( $urlToUse, 'GET', $services, $config );
+		}
+
+		return $status;
+	}
+
+	private static function getHttpStatus( $url, $method, $services, $config ) {
 		$httpProxy = $config->get( 'RottenLinksHTTPProxy' );
 
+		$userAgent = $config->get( 'RottenLinksUserAgent' );
```

Comment:
Can this just use:
```lang=php
$userAgent = <config> ?:
	<default>;
```
?
The ?: will support null, false, empty string, etc...

Reply:
Sure! I'm a PHP noob and I didn't know that operator existed. I'll push a commit shortly.